src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -356,6 +356,10 @@ private CommandResult executeCommand(String commandText) throws CommandException
 
             if (commandResult.getMode() != null) {
                 switchMode(commandResult.getMode());
+
+                if (commandResult.getMode() == ModeEnum.DASHBOARD) {
+                    projectCompletionStatusPanel.updateOcp();
```

Comment:
personally I feel that there is no harm in updating with each command. same goes for calendar
maybe can merge master into your branch because some of the parts you edited were changed today

Reply:
If we allow update for with each command, does that mean the updateOcp() method will be called if user input `help` in dashboard mode? Because if yes, the OCP will animate again from 0%. And that would look strange. So I prefer to keep the method called only when we `switch db` or click on the dashboard tab for GUI. Everything else should not restart the animation. 😨